### PR TITLE
pipe storage du implementation

### PR DESCRIPTION
**Pull request by ekazachkova**
This PR provides implementation for issue #548 

The following changes were added:
- a new API method `GET datastorage/path/usage` that calculates datastorage usage statistics. The data is taken from elasticsearch.
- method for loading NFS datastorages by name or ID was fixed.
- a new CLI option `pipe storage du` was added. If `path` is not specified all buckets will be listed with `GET datastorage/path/usage` API method call. If path was specified and it's not NFS the bucket will be listed with cloud provider directly. If path has NFS type the data will be loaded with `GET datastorage/path/usage` API method call

The command  `pipe storage du` has the following options:
- `name` - the data storage name [optional]
- `-p (--relative-path)` - the relative path [optional]
- `-f (--format)` - the size unit format [default: Mb]
- `-d (--depth)` - the maximum depth level [optional]

Usage examples:
- `pipe storage du` - lists all storages usage
```
  Storage                 Files count     Size (Mb)  
  storage_1                    26863          357.7  
  storage_2                        7            0.1  
```
- `pipe storage du storage_1` - prints usage for required storage
```
  Storage                              Files count     Size (Mb)  
  storage_1                                  26863         357.7  
```
- `pipe storage du storage_1 -p common` - prints usage for required storage and path
```
  Storage                              Files count     Size (Mb)  
  storage_1/common                           26863         357.7  
```
- `pipe storage du storage_1 -p common -d 1` - prints usage for required storage and path for depth level 1
```
  Storage                              Files count     Size (Mb)  
  storage_1/common/folder1                   26863         357.7  
  storage_1/common/folder2                       7           0.1  
```
- `pipe storage du storage_1 -f G` - prints usage for required storage with specified size format
```
  Storage                              Files count     Size (Gb)  
  storage_1                                  26863           2.2 
```

Note: `--depth` option is not supported for NFS storages now

### CODE REVIEW

File: pipe-cli/pipe.py [hunk 1]
```diff
@@ -901,6 +902,16 @@ def storage_copy_item(source, destination, recursive, force, exclude, include, q
                              exclude, include, quiet, tags, file_list, symlinks, skip_existing=skip_existing)
 
 
+@storage.command('du')
+@click.argument('path', required=False)
+@click.option('-f', '--format', help='Format for size [G/M/K]',
+              type=click.Choice(DuFormatType.possible_types()), required=False, default='M')
+@click.option('-d', '--depth', help='Depth level', type=int, required=False)
+@Config.validate_access_token(quiet_flag_property_name='quiet')
+def storage_copy_item(path, format, depth):
```
Comment (mzueva) [inline pipe-cli/pipe.py]: It seems that method name is misleading

File: pipe-cli/src/utilities/storage/s3.py [hunk 2]
```diff
@@ -411,6 +414,83 @@ def list_items(self, relative_path=None, recursive=False, page_size=StorageOpera
...
+            if 'Contents' in page:
+                for file in page['Contents']:
+                    name = self.get_file_name(file, prefix, True)
+                    size = file['Size']
+
+                    tokens = name.split(delimiter)
+                    root.data.add_item(size)
+
+                    if len(tokens) - relative_path_len > 1:
+                        if relative_path:
```
Comment (mzueva) [inline pipe-cli/src/utilities/storage/s3.py]: Could we extract this logic into a separate method to make it more readable?

File: api/src/main/java/com/epam/pipeline/manager/security/GrantPermissionManager.java [hunk 3]
```diff
@@ -412,6 +412,15 @@ public boolean storagePermission(Long storageId, String permissionName) {
         }
     }
 
+    public boolean storagePermissionByName(final String identifier, final String permissionName) {
+        AbstractSecuredEntity storage = entityManager.loadByNameOrId(AclClass.DATA_STORAGE, identifier);
```
Comment (mzueva) [inline api/src/main/java/com/epam/pipeline/manager/security/GrantPermissionManager.java]: missing `final`

**Review by mzueva — CHANGES_REQUESTED** [review]

**Comment by mzueva** [comment]
@ekazachkova Could you please add nfs storage path format to the PR description?

File: api/src/main/java/com/epam/pipeline/manager/security/GrantPermissionManager.java (hunk 3 above)
Comment (mzueva) [reply api/src/main/java/com/epam/pipeline/manager/security/GrantPermissionManager.java]: still missing `final`

**Review by mzueva — CHANGES_REQUESTED** [review]

**Comment by ekazachkova** [comment]
Several remarks for this implementation:
1. It would be better to specify storage path with single option, e.g. `pipe storage du s3://storage_1/folder1`. But in this case it's not clear how to parse NFS storage path since it is difficult to determine where the storage name and where relative path. 
The idea is to implement NFS path parsing into the server part:
- determine deepest folder path from specified path
- try to load storage by this name
- if the storage does not exists - level up folder path and try to load storage by new name
- repeat until the storage is found
2. `--depth` option is not supported for NFS storages. One of the possible ways to solve this problem is to use elasticsearch. The idea is to store depth information for paths.

**Pull request merged**